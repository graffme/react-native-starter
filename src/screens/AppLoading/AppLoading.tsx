/**
 * Loader is displayed while app is loading, state is rehydrated.
 * TODO Loads all required assets.
 * TODO Handle Splash screen.
 */
import 'react-native-gesture-handler';
import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const AppLoading: React.FunctionComponent<any> = () => {
  /* TODO Add logic for handling state here and switch routes */

  return (
    <View style={styles.container}>
      <Text>App Loading</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
});

export default AppLoading;
