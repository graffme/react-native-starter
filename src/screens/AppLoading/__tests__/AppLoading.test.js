import React from 'react';
import AppLoading from '../AppLoading';
import { shallow } from 'enzyme';

describe('AppLoading', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<AppLoading />);
    expect(wrapper).toMatchSnapshot();
  });
});
