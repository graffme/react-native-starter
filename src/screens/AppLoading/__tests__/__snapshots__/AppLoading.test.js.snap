// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoading should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    App Loading
  </Text>
</View>
`;
